lib/paper_trail/reifiers/has_one.rb

```diff
@@ -74,7 +74,18 @@ def load_version(assoc, model, transaction_id, version_at)
           when 1
             versions.first
           else
-            raise FoundMoreThanOne.new(base_class_name, versions.length)
+            case PaperTrail.config.association_reify_error_behaviour.to_s
+            when "warn"
+              version = versions.first
+              version.logger&.warn(
+                format(FoundMoreThanOne::MESSAGE_FMT, base_class_name, versions.length)
+              )
```

Comment:
Nice idea to use `version.logger`.

I think I'd prefer:

```ruby
version.logger&.warn(
  FoundMoreThanOne.new(base_class_name, versions.length).message
)
```

What do you think?

Reply:
Meh. I'd prefer not to because that creates an extra error object for no reason.

Reply:
> Meh. I'd prefer not to because that creates an extra error object for no reason.

The reason is to avoid duplicating the implementation of `FoundMoreThanOne#message`. By duplicating the implementation, if the message changes, you have to change it in two places.

> .. that creates an extra error object ..

For me, the duplication of code seems worse than a few object allocations, but that's not based on any benchmark of course, just a gut instinct.